static/client/register/js/register.js

```diff
@@ -0,0 +1,116 @@
+window.matrixRegistration = {
+    endpoint: location.origin + "/_matrix/client/api/v1/register"
+};
+
+var setupCaptcha = function() {
+    if (!window.matrixRegistrationConfig) {
+        return;
+    }
+    $.get(matrixRegistration.endpoint, function(response) {
+        var serverExpectsCaptcha = false;
+        for (var i=0; i<response.flows.length; i++) {
+            var flow = response.flows[i];
+            if ("m.login.recaptcha" === flow.type) {
+                serverExpectsCaptcha = true;
+                break;
+            }
+        }
+        if (!serverExpectsCaptcha) {
+            console.log("This server does not require a captcha.");
+            return;
+        }
+        console.log("Setting up ReCaptcha for "+matrixRegistration.endpoint);
+        var public_key = window.matrixRegistrationConfig.recaptcha_public_key;
+        if (public_key === undefined) {
+            console.error("No public key defined for captcha!");
+            return;
```

Comment:
I understand that it indicates a misconfigured server, but I think its quite likely that server admins will forget to set the registration config and not test the fallback registration page. It would be nicer if the end user of an app saw an error message along the lines of "Registration has not been correctly configured, please contact the server admins", rather than it just silently failing.

This is exacerbated by the fact that there are multiple config files to update when enabling CAPTCHAs.